### PR TITLE
Migrate CircleCI to GithubActions

### PR DESCRIPTION
**Pull request by asachdeva**
This feature branch is an attempt to migrate CircleCI to Github Actions workflows

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Build
+
+on: [push]
```
Comment (gvolpe) [inline .github/workflows/ci.yml]: You need to make this build on PRs as well like [here](https://github.com/profunktor/redis4cats/blob/master/.github/workflows/ci.yml#L7). Right now it's not building.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Build
+
+on:
+  push:
```
Comment (gvolpe) [inline .github/workflows/ci.yml]: Add `master` branch like [here](https://github.com/profunktor/redis4cats/blob/master/.github/workflows/ci.yml#L7)

Comment (asachdeva) [reply .github/workflows/ci.yml]: okay...done as requested

**Comment by asachdeva** [comment]
Just realized that I will also need to update the .mergify.yml.  I will update that later today

**Comment by gvolpe** [comment]
I wonder why the GH action is not triggered :thinking:

**Comment by asachdeva** [comment]
It was being triggered in my forked repo for sure...https://github.com/asachdeva/http4s-tracer/actions.  Thats one of the reasons why I had yanked the branches option so it would also trigger my feature branch.

**Comment by gvolpe** [comment]
Right, that's fine for testing but I don't want the CI build to be triggered on a every push once this is merged :)

Still wondering why it is not triggered as a PR build like this: https://github.com/profunktor/http4s-jwt-auth/pull/44

**Comment by gvolpe** [comment]
I've testing the caching layer in this other PR (took it from your code). You can bring 'em back :)

After the mergify file is updated, I'll merge it even if the build doesn't trigger here :)

**Comment by asachdeva** [comment]
Maybe since it was local to your repo to begin with and not a forked copy as in the PR itself was issued by you who owns that repo?   Understood on your previous comment.  I just needed a way to test it though.

**Comment by asachdeva** [comment]
> I've testing the caching layer in this other PR (took it from your code). You can bring 'em back :)
> 
> After the mergify file is updated, I'll merge it even if the build doesn't trigger here :)

Sounds good.

**Comment by gvolpe** [comment]
Yeah maybe it's that...

Sure, feel free to remove that branch naming to test it on your fork and once you get it just bring those two lines back and it'll be ready to merge :muscle:

**Comment by asachdeva** [comment]
> Yeah maybe it's that...
> 
> Sure, feel free to remove that branch naming to test it on your fork and once you get it just bring those two lines back and it'll be ready to merge muscle

I will do so after mergify update as well.  Breaking some new ground there.

**Comment by asachdeva** [comment]
Done and done

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: olafurpg/setup-scala@v5
+
+    - name: Cache Coursier dependencies
+      uses: actions/cache@v1
+      with:
+       path: ~/.cache/coursier
+       key: ${{ runner.os }}-${{ github.ref }}
```
Comment (gvolpe) [inline .github/workflows/ci.yml]: I found this name collides with the other cache, they need different names. See https://github.com/profunktor/http4s-jwt-auth/commit/b4965c94bd9a32f7e474ea98a1899df9e3d61d08

Comment (asachdeva) [reply .github/workflows/ci.yml]: Updated accordingly, however with Sbt 1.3.x the .ivy2 resolution now uses coursier behind the scenes as the dependency management repo by default, so I think we only need the .ivy2 to be cached unless I am missing something.

**Comment by gvolpe** [comment]
Thanks for your patience @asachdeva ! :tada:

**Pull request merged**